src/provider.ts

```diff
@@ -176,9 +175,11 @@ export function JsonRpcApiProvider<TBase extends Constructor<ethers.JsonRpcApiPr
         async l1TokenAddress(token: Address): Promise<string> {
             if (token == ETH_ADDRESS) {
                 return ETH_ADDRESS;
+            } else if (token == this.contractAddresses().baseToken) {
```

Comment:
```
if (token == ETH_ADDRESS) {
   
  // check if base token is ETH or not, it its not then return base token address, otherwise return ETH`
 ```

Reply:
ETH_ADDRESS is actually equal to the system contract address that manage the base token

On L1, its address is either native ETH (no address) or l1 base token address if the chain is configure to use custom base token

Reply:
fixed